include/ada/url.h

```diff
@@ -366,6 +366,16 @@ namespace ada {
      */
     [[nodiscard]] bool parse_opaque_host(std::string_view input);
 
+    /**
+     * @private
+     * 
+     * Sets the host or hostname according to override condition.
+     * Return true on success.
+     * @see https://url.spec.whatwg.org/#hostname-state
+     */
+
+    [[nodiscard]] bool set_host_or_hostname(std::string_view input, bool override_hostname);
```

Comment:
Please move this function below the `setters` in this file.

Reply:
have moved it
also, one question would it also not make this function not private? is that ok? (my cpp knowledge isnt the sharpest 😅😅)